ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/VectorUDFMapIndexStringScalar.java

```diff
@@ -66,10 +66,13 @@ public int findScalarInMap(MapColumnVector mapColumnVector, int mapBatchIndex) {
     byte[][] keyVector = keyColVector.vector;
     int[] keyStart = keyColVector.start;
     int[] keyLength = keyColVector.length;
+    final boolean isRepeating = keyColVector.isRepeating;
     for (int i = 0; i < count; i++) {
       final int keyOffset = offset + i;
+      final int len = isRepeating? keyLength[i]: keyLength[keyOffset];
```

Comment:
Is that supposed to be [i] or [0]

The two rows look a bit wrong there.

I would write keyOffset = isRepeating ? 0 : offset + i

if the intention was to use =0 for all repeating.

Reply:
I found that the key is set isRepeating only when there is a single key (count=1), but I am not sure if different orc versions may have multiple keys, if there are multiple keys in repeating, keyLength may have multiple entries.  Would it be better just deal with the isRepeating and count=1 separately?

Reply:
Agree with Gopal. If isRepeating=true, then keyLength and keyVector will have to be looked at index = 0. I was also expecting something like below
```
final int len = isRepeating? keyLength[0]: keyLength[keyOffset];
```

Reply:
Thank you for the review @prasanthj ! updated.  I only see isRepeating is true for one single key, but not sure if there can be multiple keys and still have isRepeating=true

Reply:
Could you update the test case with multiple keys and see if that produces expected result?

Reply:
added test cases of repeating multiple keys and mixed keys.

Reply:
@prasanthj appreciate your review!